07_StagingAndMultipleQueues/main.cpp

```diff
@@ -73,7 +84,414 @@ class StagingAndMultipleQueuesApp final : public examples::BasicMultiQueueApplic
 			return retval;
 		}
 
+		template<typename... Args>
+		void logFailAndTerminate(const char* msg, Args&&... args)
+		{
+			m_logger->log(msg, ILogger::ELL_ERROR, std::forward<Args>(args)...);
+			std::exit(-1);
+		}
```

Comment:
can move this to the application framework that declares `logFailure`

but make sure you take an exit code as the first argument before the message